tests/components/nexia/test_config_flow.py

```diff
@@ -80,6 +80,67 @@ async def test_form_cannot_connect(hass):
     assert result2["errors"] == {"base": "cannot_connect"}
 
 
+async def test_form_invalid_auth_http_401(hass):
+    """Test we handle invalid auth error from http 401."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    response_mock = MagicMock()
+    type(response_mock).status_code = 401
+    with patch(
+        "homeassistant.components.nexia.config_flow.NexiaHome.login",
+        side_effect=HTTPError(response=response_mock),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_USERNAME: "username", CONF_PASSWORD: "password"},
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "invalid_auth"}
+
+
+async def test_form_cannot_connect_not_found(hass):
+    """Test we handle cannot connect from an http not found error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    response_mock = MagicMock()
+    type(response_mock).status_code = 404
+    with patch(
+        "homeassistant.components.nexia.config_flow.NexiaHome.login",
+        side_effect=HTTPError(response=response_mock),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_USERNAME: "username", CONF_PASSWORD: "password"},
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_broad_exception(hass):
+    """Test we handle invalid auth error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.nexia.config_flow.NexiaHome.login",
+        side_effect=ValueError,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_USERNAME: "username", CONF_PASSWORD: "password"},
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "unknown"}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6103615)